### PR TITLE
main menu: add disabled and checked state for menu items

### LINKED ISSUE #1753: MenuItem support Selection Indication
### Is your feature request related to a problem? Please describe:
MenuItem cannot indicate that it is currently selected.

### Is it possible to construct a solution with the existing API?
no

### Describe the solution you'd like to see:
A way to indicate that a MenuItem within a Menu or PopUp is selected.

The main use case I have for this is showing which option within a PopUp is currently selected eg. the currently selected option of a Select or SelectEntry, or the current tab in an AppTab overflow PopUp or DocTab all tabs PopUp.

### PR DESCRIPTION
**Pull request by tehsphinx**
### Description:
This draft / PR is intended to add enabling/disabling of menu items as well as adding the ability to add a check before a menu item.

Fixes #1753
Addresses Enable/Disable part of #682

### Checklist:

- [ ] Tests included. (objectiveC tests missing)
- [x] Lint and formatter run with no errors.
- [ ] Tests all pass.
- [ ] Public APIs match existing style. (to be discussed)

### CODE REVIEW

**Comment by tehsphinx** [comment]
Would be happy about some help with tests, especially the objectiveC tests as I don't know objectiveC at all.

File: widget/menu_item.go [hunk 1]
```diff
@@ -245,6 +271,24 @@ func (r *menuItemRenderer) Refresh() {
 		r.background.Hide()
 	}
 	r.background.Refresh()
+
+	if r.i.Item.Disabled {
+		r.text.Color = theme.DisabledColor()
+		// TODO: enable after fixing translucence and alpha channel (https://github.com/fyne-io/fyne/issues/1977)
```
Comment (andydotxyz) [inline widget/menu_item.go:277]: This is not the right way - it should be changed to a DisabledResource and the colour will update appropriately.
You could look at the existing `Button` widget that disables icons as appropriate.

File: widget/menu_item.go [hunk 2]
```diff
@@ -245,6 +271,24 @@ func (r *menuItemRenderer) Refresh() {
...
+		// r.checkIcon.Translucency = 0.7
+	} else {
+		r.text.Color = theme.ForegroundColor()
+		r.checkIcon.Translucency = 0
```
Comment (andydotxyz) [inline widget/menu_item.go:281]: Shouldn't be needed - see above, it will be toggling disabled type for the resource instead.

**Review by andydotxyz — CHANGES_REQUESTED** [review]
This looks really great thanks.
Please merge develop in to fix the conflict and address the TODO which has a solution available already.

**Comment by prifre** [comment]
I wrote on slack: 
"I am looking for how to disable/enable a MenuItem...
I looked at https://github.com/fyne-io/fyne/pull/1975 but I cannot implementation...
What is the status?"
I got reply from tehsphinx  13:32
"fyne has made some decisions that make it unsuitable for our needs, so we moved on to other alternatives."

**Comment by andydotxyz** [comment]
I will try to pick this up myself to follow the PR in to landing.

**Comment by andydotxyz** [comment]
Thanks for your hard work on this @tehsphinx, I have resolved the merge conflicts and fixed a couple of issues noted above to hopefully land this cool feature - #2261.

**Pull request closed without merging**